update-go-version.sh

```diff
@@ -17,9 +17,9 @@ sed -i "s,^FROM registry.access.redhat.com/ubi8/go-toolset:[.0-9]\+,FROM registr
 sed -i "s,^FROM registry.access.redhat.com/ubi8/go-toolset:[.0-9]\+,FROM registry.access.redhat.com/ubi8/go-toolset:${golang_base_version}," images/*/Containerfile
 for f in .github/workflows/*.yml; do
     if [ $(yq  eval '.jobs.build.strategy.matrix | has("go")' "$f") == "true" ]; then
-      yq eval --inplace ".jobs.build.strategy.matrix.go[0] = ${golang_base_version} | .jobs.build.strategy.matrix.go[0] style=\"single\"" "$f";
+      yq eval --inplace ".jobs.build.strategy.matrix.go[0] = \"${golang_base_version}\" | .jobs.build.strategy.matrix.go[0] style=\"single\"" "$f";
```

Comment:
With the latest changes in this PR, `golang_base_version` is `1.21`, not `1.21.0` as in the commit log, this is no longer needed (at least according to my testing with yq 4.43.1
The changes in this commit can apparently be simplified to:
```
 yq eval --inplace ".jobs.build.strategy.matrix.go[0] = \"${golang_base_version}\"" "$f";
```

Reply:
it is working because we have already single quote around the version info, if we remove it on one of that then it will just use the double quote with `single` style we are explicit making it to use single quote. 

yes if we are using `1.21` only then this is not needed.